### PR TITLE
upgrade k8s/api to v0.21.1 (in apis/go.mod) and controller-tools to v0.6.0

### PR DESCRIPTION
**Pull request by asalkeld**
This also required an upgrade of controller-runtime to 0.9.0 which caused a minor unit test change.

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
Are we also planning to upgrade Controller Tools, since that is also outdated?

**Comment by asalkeld** [comment]
> Are we also planning to upgrade Controller Tools, since that is also outdated?

I can do that :+1:

File: controllers/metal3.io/baremetalhost_controller_test.go [hunk 1]
```diff
@@ -2004,8 +2005,24 @@ func TestInvalidBMHCanBeDeleted(t *testing.T) {
 
 	doDeleteHost(host, r)
 
+	seenDeletingState := false
+	seenDeleted := false
 	tryReconcile(t, r, host, func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
-		return host.Status.Provisioning.State == metal3v1alpha1.StateDeleting && len(host.Finalizers) == 0
```
Comment (andfasano) [inline controllers/metal3.io/baremetalhost_controller_test.go:2008]: @asalkeld  I guess the same problem has been addressed also in #903, see here https://github.com/metal3-io/baremetal-operator/pull/903/files#diff-20c1a976622b8d54cf98349e3e6219778fa00e89ed669f1bb12cd778326469bbR142

**Comment by dtantsur** [comment]
/hold

Please hold until https://github.com/metal3-io/baremetal-operator/pull/884 merges, it has been around for a month, and this patch is a merge conflict bomb.

**Comment by hardys** [comment]
FYI I just approved #903 which includes the controller-runtime update

**Comment by asalkeld** [comment]
> FYI I just approved #903 which includes the controller-runtime update

thanks @hardys , I have rebased. The change to apis/go.mod was missing. This also includes the controller-tools upgrade..

**Pull request merged**